app/docker/views/images/import/importImageController.js

```diff
@@ -1,28 +1,63 @@
+import { PorImageRegistryModel } from 'Docker/models/porImageRegistry';
+
 angular.module('portainer.docker').controller('ImportImageController', [
   '$scope',
   '$state',
   'ImageService',
   'Notifications',
   'HttpRequestHelper',
-  function ($scope, $state, ImageService, Notifications, HttpRequestHelper) {
+  'Authentication',
+  'ImageHelper',
+  'endpoint',
+  function ($scope, $state, ImageService, Notifications, HttpRequestHelper, Authentication, ImageHelper, endpoint) {
     $scope.state = {
       actionInProgress: false,
     };
 
+    $scope.endpoint = endpoint;
+
+    $scope.isAdmin = Authentication.isAdmin();
+
     $scope.formValues = {
       UploadFile: null,
       NodeName: null,
+      RegistryModel: new PorImageRegistryModel(),
     };
 
+    $scope.setPullImageValidity = setPullImageValidity;
+    function setPullImageValidity(validity) {
+      $scope.state.pullImageValidity = validity;
+    }
+
+    function tagImage(id) {
+      const registryModel = $scope.formValues.RegistryModel;
+      if (registryModel.Image) {
+        const image = ImageHelper.createImageConfigForContainer(registryModel);
+        ImageService.tagImage(id, image.fromImage).catch(function error(err) {
+          Notifications.error('Failure', err, 'Unable to tag image');
+        });
+      }
+    }
+
     $scope.uploadImage = function () {
       $scope.state.actionInProgress = true;
 
       var nodeName = $scope.formValues.NodeName;
       HttpRequestHelper.setPortainerAgentTargetHeader(nodeName);
       var file = $scope.formValues.UploadFile;
       ImageService.uploadImage(file)
-        .then(function success() {
-          Notifications.success('Images successfully uploaded');
+        .then(async function success(res) {
```

Comment:
can you refactor the `uploadImage` function to use async/await, `const` instead of `var`?

Reply:
Do you mean removing then/catch/finally and assigning the result/error to const? something like below.

```
      const [res, err] = await ImageService.uploadImage(file);
      if (err != null) {
        Notifications.error('Failure', err, 'Unable to upload image');
      }

      if (res.data.stream) {
        const regex = /Loaded.*?: (.*?)\n$/g;
        const imageIds = regex.exec(res.data.stream);
        if (imageIds && imageIds.length == 2) {
          await tagImage(imageIds[1]);
          $state.go('docker.images.image', { id: imageIds[1] }, { reload: true });
        }
        Notifications.success('Images successfully uploaded');
      } else {
        Notifications.success('The uploaded tar file contained multiple images. The provided tag therefore has been ignored.');
      }

      $scope.state.actionInProgress = false;
```

Reply:
image upload api return 200 response in case of error so the the current catch is not working.

![image](https://user-images.githubusercontent.com/91852476/139601474-54442b5b-538b-4fe2-bc10-3718fcc96ee6.png)

Reply:
nope, refactor the whole function to use async await, just like you did with the success function.

Reply:
done

Reply:
```suggestion
       const {data} = await ImageService.uploadImage(file)
       if (data.error) {
          ...
        }
        
        if (data.stream) {
           ...
         }
```

Reply:
done please review